app/models/taxon.rb

```diff
@@ -68,4 +74,13 @@ def can_subscribe?
 
     true
   end
+
+private
+
+  def fetch_tagged_content
+    taxon_content_ids = [content_id] + associated_taxons.map(&:content_id)
+    taxon_content_ids.inject([]) do |content, taxon_content_id|
+      content + TaggedContent.fetch(taxon_content_id)
```

Comment:
Might be worth exploring whether Rummager allows multiple `content_id`'s in a single request for `filter_taxons` - it looks like it is passing in an array (`filter_taxons: [content_id],`) - https://github.com/alphagov/collections/blob/master/app/services/tagged_content.rb#L30